ros/angel_system_nodes/angel_system_nodes/base_emotion_detector.py

```diff
@@ -0,0 +1,192 @@
+            self.listener_callback,
+            1,
+        )
+        self._interp_emo_publisher = self.create_publisher(
+            InterpretedAudioUserEmotion, self._out_interp_uemotion_topic, 1
+        )
+
+        self.message_queue = queue.Queue()
+        self.handler_thread = threading.Thread(target=self.process_message_queue)
+        self.handler_thread.start()
+
+        self.sentiment_analysis_model = SentimentIntensityAnalyzer()
+
+    def _get_vader_sentiment_analysis(self, utterance: str):
+        """
+        Applies Vader Sentiment Analysis model to assign 'positive,' 'negative,'
+        and 'neutral' sentiment labels. Returns with  a 100% confidence.
+        """
+        polarity_scores = self.sentiment_analysis_model.polarity_scores(utterance)
+        if polarity_scores["compound"] >= VADER_POSITIVE_COMPOUND_THRESHOLD:
+            classification = LABEL_MAPPINGS["pos"]
+        elif polarity_scores["compound"] <= VADER_NEGATIVE_COMPOUND_THRESHOLD:
+            classification = LABEL_MAPPINGS["neg"]
+        else:
+            classification = LABEL_MAPPINGS["neu"]
+
+        confidence = 1.00
+        colored_utterance = colored_utterance = colored(utterance, "light_blue")
+        colored_emotion = colored(classification, "light_green")
+        self.log.info(
+            f'Rated user utterance:\n>>> "{colored_utterance}"'
+            + f"\n>>> with emotion scores {polarity_scores}.\n>>> "
+            + f'Classifying with emotion="{colored_emotion}" '
+            + f"and score={confidence}"
+        )
+        return (classification, confidence)
+
+    def get_inference(self, msg):
+        """
+        Abstract away the different model inference calls depending on the
+        node's configure model mode.
+        """
+        return self._get_vader_sentiment_analysis(msg.utterance_text)
+
+    def listener_callback(self, msg):
+        """
+        This is the main ROS node listener callback loop that will process
+        all messages received via subscribed topics.
+        """
+        self.log.debug(f'Received message:\n\n"{msg.utterance_text}"')
+        if not self._apply_filter(msg):
+            return
+        self.message_queue.put(msg)
+
+    def process_message_queue(self):
+        """
+        Constant loop to process received messages.
+        """
+        while True:
+            msg = self.message_queue.get()
+            self.log.debug(f'Processing message:\n\n"{msg.utterance_text}"')
+            classification, confidence_score = self.get_inference(msg)
+            self.publish_detected_emotion(
+                msg.utterance_text, classification, confidence_score
+            )
+
+    def publish_detected_emotion(
+        self, utterance: str, classification: str, confidence_score: float
+    ):
+        """
+        Handles message publishing for an utterance with a detected emotion classification.
+        """
+        emotion_msg = InterpretedAudioUserEmotion()
```

Comment:
Can you also fill out the `header` field here? Could fill in the timestamp and the frame_id fields

Reply:
Oh  yes. Similarly to  PR#296. Will  do  that.